projects/packages/my-jetpack/_inc/state/selectors.js

```diff
@@ -143,6 +143,37 @@ const getStatsCounts = state => {
 	return state.statsCounts?.data;
 };
 
+const getUserStats = state => {
+	return state.userStats;
+};
+
+const getGuessedSiteLifecycleStatus = state => {
+	const { modules, purchases, features, plugins, isSiteConnected, isUserConnected } =
+		getUserStats( state );
+
+	// 'new' = no purchases + less than 3 features + less than 3 modules
+	if ( purchases.length === 0 && features.length < 3 && modules.length < 3 ) {
+		// 'brand-new' = 'new' + (no user or site connection + no modules + only one plugin)
+		if (
+			( ! isUserConnected || ! isSiteConnected ) &&
+			modules.length === 0 &&
+			plugins.length === 1
+		) {
+			return 'brand-new';
+		}
+
+		return 'new';
+	}
+
+	// 'settling-in' = 1 purchase and less than 10 features
+	if ( purchases.length === 1 && features.length < 10 ) {
```

Comment:
Related to the comment above, I think this can be a modules check instead of a features check